canary/producer-c/jobs/runner.groovy

```diff
@@ -173,12 +162,33 @@ pipeline {
         booleanParam(name: 'USE_IOT')
     }
 
+    // Set the role ARN to environment to avoid string interpolation to follow Jenkins security guidelines.
+    environment {
+        AWS_KVS_STS_ROLE_ARN = credentials('CANARY_STS_ROLE_ARN')
```

Comment:
let us check that the AWS_KVS_CANARY_ROLE_ARN has least privileges only for kinesis ops

 aws iam simulate-principal-policy \
                    --policy-source-arn $AWS_KVS_CANARY_ROLE_ARN \
                    --action-names "kinesisvideo:*" \..
                    
    add credentials plugin if we havenot to mask the creds in the logs and binding works correctly in the  env.
                    
                    https://plugins.jenkins.io/credentials/